src/app/lib/views/torrent_collection.js

```diff
@@ -76,179 +107,193 @@
+                                        peers: ldata.leechers,
+                                        size: ldata.size,
+                                        index: index
+                                    };
+                                    indx++;
+                                    if (item.title.match(/trailer/i) !== null && input.match(/trailer/i) === null) {
+                                        return;
+                                    }
+                                    results.push(itemModel);
+                                    index++;
+                                }).catch(function (err) {
+                                    throw 'nope';
+                                });
+                            });
+                        }).catch(function (err) {
+                            console.error('1337x search:', err);
+                            resolve(results);
+                        });
+                    });
+                }
+            };
+
+            var rarbg = function (retry) {
+                if (Settings.enablerarbg) {
+                    return new Promise(function (resolve) {
+                        var results1 = [];
+                        setTimeout(function () {
+                            resolve(results1);
+                        }, 6000);
+                        var rbg = torrentCollection.rbg;
+                        rbg.search({
+                            query: input.toLocaleLowerCase(),
+                            category: category.toLocaleLowerCase() === 'movies' ? 'movies' : 'tv',
+                            sort: 'seeders',
+                            verified: false
+                        }).then(function (results) {
+                            console.debug('rarbg search: %s results', results.length);
+                            results.forEach(function (item) {
                                 var itemModel = {
-                                    title: ldata.title,
-                                    magnet: ldata.download.magnet,
-                                    seeds: ldata.seeders,
-                                    peers: ldata.leechers,
-                                    size: ldata.size,
+                                    title: item.title,
+                                    magnet: item.download,
+                                    seeds: item.seeders,
+                                    peers: item.leechers,
+                                    size: Common.fileSize(parseInt(item.size)),
                                     index: index
                                 };
-                                indx++;
+
                                 if (item.title.match(/trailer/i) !== null && input.match(/trailer/i) === null) {
                                     return;
                                 }
-                                results.push(itemModel);
+                                results1.push(itemModel);
```

Comment:
why is it results1?

Reply:
Not me!!! :D  

I see what happened.. (though I did write a note about it). These arent changes. I didnt change anything in those function. I just added an if statement for each of them, so I had to also indent. I also re-arranged them so they are tpb, 1337x, rarbg,omgtorrents and have the same order as everywhere else. Other than that all the code they contain is pre-existing code. Github just crapped itselfs a bit trying to figure out the edits. Sorry :)

Reply:
oh, that makes sense. of course I totally missed the note, my bad :) this file should definitely get a rewrite eventually, but that's out of the scope of this PR obviously

Reply:
I do still need to camelCase some names. Also if you have any suggestions renaming some functions or variables to avoid confusion let me know and I'll edit them.